libsql/src/statement.rs

```diff
@@ -92,18 +93,37 @@ pub struct MappedRows<F> {
     map: F,
 }
 
-impl<F, T> Iterator for MappedRows<F>
+impl<F> MappedRows<F> {
+    pub fn new(rows: Rows, map: F) -> Self {
+        Self { rows, map }
+    }
+}
```

Comment:
Overall, I think it would be a lot easier to have a `into_stream` method on Rows, and then use `Stream`'s provided methods for mapping etc

Reply:
it would look something like:
```rust
fn into_stream(self) -> impl Stream<...> {
    async_stream! {
        yield self.next().await
     }
}
```